Sources/Hedera/Account/AccountInfoQuery.swift

```diff
@@ -59,12 +59,48 @@ public final class AccountInfoQuery: Query<AccountInfo> {
         try await Proto_CryptoServiceAsyncClient(channel: channel).getAccountInfo(request)
     }
 
-    internal override func makeQueryResponse(_ response: Proto_Response.OneOf_Response) throws -> Response {
+    internal override func makeQueryResponse(_ context: Context, _ response: Proto_Response.OneOf_Response) async throws
+        -> Response
+    {
+        let mirrorNodeGateway = try MirrorNodeGateway.forNetwork(context.mirrorNetworkNodes, context.ledgerId)
+        let mirrorNodeService = MirrorNodeService.init(mirrorNodeGateway)
+
         guard case .cryptoGetInfo(let proto) = response else {
             throw HError.fromProtobuf("unexpected \(response) received, expected `cryptoGetInfo`")
         }
 
-        return try .fromProtobuf(proto.accountInfo)
+        let accountInfo = try AccountInfo.fromProtobuf(proto.accountInfo)
+        let tokenRelationshipsProto = try await mirrorNodeService.getTokenRelationshipsForAccount(
+            String(describing: accountInfo.accountId.num))
+
+        var tokenRelationships: [TokenId: TokenRelationship] = [:]
+
+        for relationship in tokenRelationshipsProto {
+            tokenRelationships[.fromProtobuf(relationship.tokenID)] = try TokenRelationship.fromProtobuf(relationship)
+        }
+
+        return AccountInfo(
+            accountId: accountInfo.accountId,
+            contractAccountId: accountInfo.contractAccountId,
+            isDeleted: accountInfo.isDeleted,
+            proxyAccountId: accountInfo.proxyAccountId,
+            proxyReceived: accountInfo.proxyReceived,
+            key: accountInfo.key,
+            balance: accountInfo.balance,
+            sendRecordThreshold: accountInfo.sendRecordThreshold,
+            receiveRecordThreshold: accountInfo.receiveRecordThreshold,
+            isReceiverSignatureRequired: accountInfo.isReceiverSignatureRequired,
+            expirationTime: accountInfo.expirationTime,
+            autoRenewPeriod: accountInfo.autoRenewPeriod,
+            accountMemo: accountInfo.accountMemo,
+            ownedNfts: accountInfo.ownedNfts,
+            maxAutomaticTokenAssociations: accountInfo.maxAutomaticTokenAssociations,
+            aliasKey: accountInfo.aliasKey,
+            ethereumNonce: accountInfo.ethereumNonce,
+            tokenRelationships: tokenRelationships,
+            ledgerId: accountInfo.ledgerId,
+            staking: accountInfo.staking
+        )
```

Comment:
I don't like the double building, from `proto` then an extra because we can't mutate the one field, can we do something about that?